### PR TITLE
Add API Endpoints for Posts and Comments

### PR DESCRIPTION
**Pull request by axcell-0**
Description:

Introduce API endpoint to list all posts for a user: GET /api/v1/users/:user_id/posts
Implement API endpoint to list all comments for a user's post: GET /api/v1/users/:user_id/posts/:post_id/comments
Add API endpoint for users to create comments on posts: POST /api/v1/users/:user_id/posts/:post_id/comments
Automatically set the comment owner as the user creating it
Configure API to accept JSON in request payload
Implement JSON responses for consistent data format

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,4 +9,12 @@
       resources :likes, only: %i[new create]
     end
   end
+
+  namespace :api, default: {format: :json} do
+    namespace :v1 do
+      resources :posts, only: %i[index] do
+        resources :comments, only: %i[index create]
+      end
+    end
```
Comment (fobadara) [inline config/routes.rb]: [REQUIRED] Kindly ensure that posts and comments resources are wrapped by users. So comments are nested in post as you have done and posts are nested in users. If you notice you are asked to create endpoints for posts and comments that belong to a user. Nesting it like that will give use endpoints that look something like this `/api/v1/users/:user_id/posts...`

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team,
Good job so far!

Highlights!
Good code organization and working app ✔️
Descriptive README ✔️
Good first attempt ✔️ 
Use of `api` and version number in routes ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . However, you still need to fix some things.

## Status: Requires Changes
Check the comments under the review.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me @julie-ify if something is not 100% clear.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: config/routes.rb [hunk 2]
```diff
@@ -8,4 +9,18 @@
...
+      resources :comments, only: %i[index create]
+    end
+  end
+
+  namespace :api, default: {format: :json} do
+    namespace :v1 do
+      resources :posts, only: %i[index] do
+        resources :comments, only: %i[index create]
+      end
+    end
```
Comment (shahadat3669) [inline config/routes.rb]: Hello there! I noticed that although you have made some updates to your code after the previous review, it still isn't working properly. You added some route configurations between lines 13 to 17 as requested by the previous reviewer, but unfortunately, these routes have not been configured within the API scope. The API part of your code is actually between lines 19 to 24. The previous reviewer also mentioned that there should be a user route within the `v1` routes, followed by posts and comments, for example, `/api/v1/users/:user_id/posts...`. I kindly request you to update your code to ensure that it works as requested.

```suggestion
  namespace :api, default: {format: :json} do
    namespace :v1 do
      resources :users, only: [] do
        resources :posts, only: %i[index] do
          resources :comments, only: %i[index create]
        end
      end
    end
  end 
```

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi everyone, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Worked on pervious reviewer comments



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 2 above)
Comment (axcell-0) [reply config/routes.rb]: please if there is anny issues after this I need practical code to learn from it please I have tried and do many research but that's all I could find

**Review by topeogunleye — APPROVED** [review]
### Hi @axcell-0, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
-  Resolved issues highlighted by previous code reviewer :heavy_check_mark:
 
## [Optional] suggestions:
* [ ]  Please check my inline comments.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**